nixos/modules/services/development/livebook.nix

```diff
@@ -80,18 +65,11 @@ in
     systemd.user.services.livebook = {
       serviceConfig = {
         Restart = "always";
-        EnvironmentFile = cfg.environmentFile;
-        ExecStart =
-          let
-            args = lib.cli.toGNUCommandLineShell { } ({
-              inherit (cfg) port;
-              ip = cfg.address;
-              name = cfg.erlang_node_name;
-              sname = cfg.erlang_node_short_name;
-            } // cfg.options);
-          in
-            "${cfg.package}/bin/livebook server ${args}";
+        EnvironmentFile = optional (cfg.environment != null) cfg.environmentFile;
```

Comment:
shouldn't it be cfg.environmentFile != null ?

also I've never seen the environment attribute before, will it merge correctly the environmentFile and the environment attribute set ?

Reply:
> shouldn't it be cfg.environmentFile != null ?

Yes! Will fix right away.

> also I've never seen the environment attribute before, will it merge correctly the environmentFile and the environment attribute set ?

It is used in other NixOS service definitions, such as [oci-containers](https://search.nixos.org/options?channel=unstable&show=virtualisation.oci-containers.containers.%3Cname%3E.environment&from=0&size=50&sort=relevance&type=packages&query=virtualisation.oci-containers.containers.%3Cname%3E.environment), but I am happy to give it another name if desired. 

Why both `environment` and `environmentFile`? I think it makes sense to be able to divide up sensitive information, such as `LIVEBOOK_PASSWORD` and put that in `environmentFile` and non-sensitive information, such as `LIVEBOOK_PORT` and put that in `environment.

The relationship between the two arguments are defined [here](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#EnvironmentFile=). It is my understanding that `EnvironmentFile` supersedes `Environment`.

Reply:
I think the separation makes sense.

I thought it was environment with a capital e, but it's not. I've even used this attribute before in one of the modules I made, never mind.

Reply:
I'd go further and say we shouldn't have both `environment` and `environmentFile`.

The main reason to use `environmentFile` over `environment` is that you can avoid putting your password in the nix store.  But if you don't care about this, you can easily just generate a file with `pkgs.writeText` like the test does.  So, I'd propose we get rid of `environement` and give an example of using `pkgs.writeText` in the docs.

Reply:
> I'd go further and say we shouldn't have both `environment` and `environmentFile`.

I don't agree, cf. [my comment above](https://github.com/NixOS/nixpkgs/pull/283447#discussion_r1464803951). I think they can serve different purposes at the same time.

Reply:
I don't have a strong opinion either way on this, I'm happy for both of you to reach the compromise that works best for everyone.